src/treq/test/test_content.py

```diff
@@ -173,3 +184,21 @@ def test_text_content_default_encoding_no_header(self):
         self.protocol.connectionLost(Failure(ResponseDone()))
 
         self.assertEqual(self.successResultOf(d), u'\xa1')
+
+    def test_text_content_unicode_headers(self):
+        """
+        Header parsing is robust against unicode header names and values.
+        """
+        self.response.headers = Headers({
+            b'Content-Type': [
```

Comment:
Oh jeez. `.encode("utf-8")`.  I see.  Sorry about that.

As far as I can tell, HTTP headers really _are_ ASCII, but apparently a more appropriate encoding would be latin-1 (i.e. charmap) to avoid any encoding surprises. http://stackoverflow.com/questions/4400678/what-character-encoding-should-i-use-for-a-http-header#4410331

So… yes, I guess you are doing the right thing here, sorry for my earlier misreading :).